src/components/Header/HeaderSearch.jsx

```diff
@@ -38,7 +35,7 @@ export default function HeaderSearch() {
         aria-label="Search through site content"
         placeholder="Search..."
         onChange={onSearchInputChange}
-        defaultValue={q}
+        value={query}
```

Comment:
`onChange={onSearchInputChange}` will set the value of `q` state instead of `query`, we cannot change the input until press `Enter`